### DISCUSSION TITLE
Dynamic Activity

### DISCUSSION THREAD

**Discussion by rmanjusha** [elsa-workflows/elsa-core#5488] [Q&A]
Is it possible to create activities dynamically with elsa?

**Comment by glime-ai** [comment]
Yes, it is possible to create activities dynamically with Elsa. Elsa Workflows is designed to be extensible, allowing for the creation of custom activities. This can be done by composing activities dynamically based on domain models and then building and running the workflow. Elsa 3, in particular, supports this natively, allowing for the dynamic composition of activities and workflows.

For example, as mentioned in one of the GitHub discussions, you could create a composite activity and then build and run the workflow dynamically like this:

```csharp
var myCompositeActivity = new Composite
{
   Root = new Sequence
   {
      new WriteLine("Wazzaa")
   }
};

var workflow = new Workflow
{
   Root = myCompositeActivity
}

workflowRunner.RunAsync(workflow);
```

This example demonstrates how you can programmatically create a workflow with a sequence of activities in Elsa 3.

For more detailed information on creating custom activities and extending Elsa, you should refer to the official documentation and resources provided by the Elsa community.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
